elfpy/data/convert_data.py

```diff
@@ -97,7 +99,160 @@ def fetch_contract_transactions_for_block(
+    """From decoded transaction logs, we look at the log that contains the trade summary
+
+    Arguments
+    ---------
+    logs: list[dict]
+        The list of dictionaries that was decoded from `eth.get_transaction_logs`
+    block_number: BlockNumber
+        The current block number of the log
+
+    Returns
+    -------
+    list[Transaction]
+        A list of Transaction objects ready to be inserted into Postgres
+    """
+    wallet_deltas = []
+    # We iterate through the logs looking for specific events that describe the transaction
+    # We then create a WalletDelta object with their corresponding token and base deltas for
+    # each action
+    for log in logs:
+        if log["event"] == "AddLiquidity":
+            wallet_addr = log["args"]["provider"]
+            token_delta = _convert_scaled_value(log["args"]["lpAmount"])
+            base_delta = _convert_scaled_value(-log["args"]["baseAmount"])
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="LP",
+                    tokenType="LP",
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                )
+            )
+
+        elif log["event"] == "OpenLong":
+            wallet_addr = log["args"]["trader"]
+            token_delta = _convert_scaled_value(log["args"]["bondAmount"])
+            base_delta = _convert_scaled_value(-log["args"]["baseAmount"])
+            maturity_time = log["args"]["maturityTime"]
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="LONG",
+                    tokenType="LONG-" + str(maturity_time),
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                    maturityTime=maturity_time,
+                )
+            )
+
+        elif log["event"] == "OpenShort":
+            wallet_addr = log["args"]["trader"]
+            token_delta = _convert_scaled_value(log["args"]["bondAmount"])
+            base_delta = _convert_scaled_value(-log["args"]["baseAmount"])
+            maturity_time = log["args"]["maturityTime"]
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="SHORT",
+                    tokenType="SHORT-" + str(maturity_time),
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                    maturityTime=maturity_time,
+                )
+            )
+
+        elif log["event"] == "RemoveLiquidity":
+            wallet_addr = log["args"]["provider"]
+            # Two deltas, one for withdrawal shares, one for lp tokens
```

Comment:
there are 3 log args here, baseAmount, withdrawalShareAmount and lpToken

Reply:
or "we add two WalletDelta objects below, to account for the 3 separate tokens being transferred: lp tokens, base, and withdrawal shares"

Reply:
These statements are going through all log args and ignoring all other logs except for the log that states `RemoveLiquidity`. I then translate that into two different `WalletDeltas`, where each wallet delta is defined as two deltas: one of a defined custom token, and one as the change in base. `RemoveLiquidity` is formatted so that this results in 2 deltas: one for `lpToken` (which accounts for the base delta), and one for `withdrawalShares` (which explicitly sets the base delta to 0)

Reply:
Updating comment to reflect this conversation